### PR TITLE
Sanjay changed progress bar in ConstestDetailsFragment

### LINKED ISSUE #8: Added loading view for the contest details screen
**Is your feature request related to a problem? Please describe.**
Currently, there is a pretty common progress bar for the contest detailed screen while most of the screens uses a better looking progress bar. It would be great to have the same progress bar there. **Make sure that you are using the same progress bar(loading view) on the screen which is used for most of the screens.**

**Describe the solution you'd like**
This one should be an easy issue to solve, all you have to do is first remove the older progress bar and replace it with the other one. You can check out the implementation of the progress bar on different screens which you can find it here for [question screen](https://github.com/cdhiraj40/LeetDroid/blob/development/app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/question/QuestionFragment.kt)

Steps:

1. Show the loading view at the start of the screen
2. Hide it once the contributor list has been fetched

**Describe alternatives you've considered**
None

**Additional context**
You can ping me in the comments if additional information is needed!

### PR DESCRIPTION
**Pull request by SanjaySargam**
### **Description**
In this PR I modified the progress bar ( i.e. loading view progress bar ) in the ContestDeatilsFragment of issue number #8 

Fixes #8

### **Screen recording**
https://user-images.githubusercontent.com/65113071/197314058-8e632007-0a90-48e9-88f2-5f93a770bf03.mp4

### Checklist:
 
- [x] My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [x]  I have commented my code, particularly in hard-to-understand areas
- [x]  My changes generate no new warnings

### CODE REVIEW

**Comment by cdhiraj40** [comment]
<p>Hey, @SanjaySargam could you please below template and update the PR description? Thanks.</p>
============


# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue_number)

## Screen recording
Add a screen recording showing the changes, if UI/UX changes are done.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

**Review by cdhiraj40 — CHANGES_REQUESTED** [review]
<p>Hey, @SanjaySargam could you please below template and update the PR description? Thanks.</p>
=============================================


# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue_number)

## Screen recording
Add a screen recording showing the changes, if UI/UX changes are done.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

**Comment by SanjaySargam** [comment]
Okkk

**Comment by cdhiraj40** [comment]
> Okkk

Let me know once you are done, I will review your PR 🚀

**Comment by SanjaySargam** [comment]
> > Okkk
> 
> Let me know once you are done, I will review your PR 🚀 

Ok fine

**Comment by SanjaySargam** [comment]
@cdhiraj40 I added PR description lets have a check

File: app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt [hunk 1]
```diff
@@ -27,8 +27,8 @@ class ContestDetailsFragment : Fragment(), ContestHistoryAdapter.OnItemClicked {
     private lateinit var fragmentContestDetailsBinding: FragmentContestDetailsBinding
     private lateinit var contestRankingJson: ContestRankingModel
     private lateinit var contestHistoryAdapter: ContestHistoryAdapter
-    private lateinit var generalErrorView: View
```
Comment (cdhiraj40) [inline app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt:30]: Why are you removing the error view? It should exist and be visible whenever an exception occurs. It should not be removed.

File: app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt [hunk 2]
```diff
@@ -38,8 +38,8 @@ class ContestDetailsFragment : Fragment(), ContestHistoryAdapter.OnItemClicked {
         // Inflate the layout for this fragment
         fragmentContestDetailsBinding = FragmentContestDetailsBinding.inflate(layoutInflater)
         val rootView = fragmentContestDetailsBinding.root
+        loadingView = rootView.findViewById(R.id.loading_view)
 
-        generalErrorView = rootView.findViewById(R.id.view_general_error)
```
Comment (cdhiraj40) [inline app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt:42]: Please undo this change.

File: app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt [hunk 3]
```diff
@@ -97,11 +97,11 @@ class ContestDetailsFragment : Fragment(), ContestHistoryAdapter.OnItemClicked {
                             contestHistoryAdapter
 
                         contestHistoryAdapter.setOnClick(this@ContestDetailsFragment)
-                        fragmentContestDetailsBinding.contestHistoryProgressBar.visibility =
-                            View.GONE
+                        loadingView.visibility = View.GONE;
+
                     }
                 } catch (exception: Exception) {
-                    generalErrorView.visibility = View.VISIBLE
```
Comment (cdhiraj40) [inline app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt:104]: Please undo this change.

File: app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt [hunk 4]
```diff
@@ -97,11 +97,11 @@ class ContestDetailsFragment : Fragment(), ContestHistoryAdapter.OnItemClicked {
...
+                    loadingView.visibility = View.VISIBLE
```
Comment (cdhiraj40) [inline app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt:104]: The loading view should not be visible when an error occurs, it should be gone and the error view should be shown which was already there.

```
                    generalErrorView.visibility = View.VISIBLE
                    loadingView.visibility = View.GONE

File: app/src/main/res/layout/fragment_contest_details.xml [hunk 5]
```diff
@@ -130,8 +134,8 @@
         android:visibility="gone" />
 
     <include
-        android:id="@+id/view_general_error"
```
Comment (cdhiraj40) [inline app/src/main/res/layout/fragment_contest_details.xml:133]: Please undo this change.

File: app/src/main/res/layout/fragment_contest_details.xml [hunk 6]
```diff
@@ -130,8 +134,8 @@
...
-        layout="@layout/general_error_layout"
+        android:id="@+id/loading_view"
+        layout="@layout/loading_screen_layout"
```
Comment (cdhiraj40) [inline app/src/main/res/layout/fragment_contest_details.xml:138]: You should add the loading view as a new view and not replace the error view.

**Review by cdhiraj40 — CHANGES_REQUESTED** [review]
Hey, @SanjaySargam I have requested some small changes. Do not remove the error view, just replace the old loading view with the new one.

File: app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt (hunk 1 above)
Comment (SanjaySargam) [reply app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt:30]: Ok

File: app/src/main/res/layout/fragment_contest_details.xml (hunk 6 above)
Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: Ok sorry for the inconvenience I will resolve this

Comment (cdhiraj40) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: No worries! Also could you please update your branch too? so that tests can pass. I have updated some of the code. Thanks.

Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: I don't know how to update branch I'm little bit new to open source. Could you please tell me?

Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: Yeah I updated the branch

Comment (cdhiraj40) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: sure just run these below commands:
```
git remote add upstream https://github.com/cdhiraj40/LeetDroid.git
git pull upstream development
```

An explanation for the commands
1. The command adds the main repository as a remote called "upstream"(you can name anything), so later you can fetch/update code right from your terminal
2. It pulls the code from the development branch of the remote upstream which is the main repo here.

Hope you understood, let me know if you need more explanation!

Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: > sure just run these below commands:
> 
> ```
> git remote add upstream https://github.com/cdhiraj40/LeetDroid.git
> git pull upstream development
> ```
> 
> An explanation for the commands
> 
> 1. The command adds the main repository as a remote called "upstream"(you can name anything), so later you can fetch/update code right from your terminal
> 2. It pulls the code from the development branch of the remote upstream which is the main repo here.
> 
> Hope you understood, let me know if you need more explanation!

Thanks for the explanation. I followed these steps but I don't know whether the branch is updated or not.

Comment (cdhiraj40) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: You already have updated the branch so no need to worry 😄

Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: Ok did you undo the required changes or have to do it?

Comment (cdhiraj40) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: no you will need to do it

Comment (SanjaySargam) [reply app/src/main/res/layout/fragment_contest_details.xml:138]: Ok

**Review by cdhiraj40 — APPROVED** [review]
Looks good to me! Thanks for your contribution 🚀
